python/rmm/_lib/device_buffer.pyx

```diff
@@ -116,6 +116,20 @@ cdef class DeviceBuffer:
         cdef DeviceBuffer other = DeviceBuffer.c_to_device(state)
         self.c_obj = move(other.c_obj)
 
+    def __setitem__(self, key, value):
+        rng = range(self.size)[key]
+        if not isinstance(rng, range):
+            rng = range(rng, rng + 1)
```

Comment:
We should probably handle negative slices / scalars here

Reply:
Negative indices are normalized by slicing the `range` object.

```python
In [1]: range(5)[-1]
Out[1]: 4
```

Currently we are raising an error for negative indices or any other step size that is not `1` since this could be a bit tricky to do in practice and it's unclear if we have use cases for those yet.